sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -287,7 +332,13 @@ private[sql] class RocksDBStateStoreProvider
       useColumnFamilies)
   }
 
-  @volatile private var encoder: RocksDBStateEncoder = _
+  private val encoderMapLock = new Object
+
+  @GuardedBy("encoderMapLock")
+  @volatile private var keyEncoderMap = new mutable.HashMap[String, RocksDBKeyStateEncoder]
+
+  @GuardedBy("encoderMapLock")
+  @volatile private var valueEncoderMap = new mutable.HashMap[String, RocksDBValueStateEncoder]
```

Comment:
Can we just use a concurrentHashMap with signature `Map[String, (RocksDBKeyStateEncoder, RocksDBValueStateEncoder)]`. Then, we don't need to synchronize using a lock.